orchestration/src/main/java/com/sap/ai/sdk/orchestration/SystemMessage.java

```diff
@@ -15,42 +14,37 @@ public class SystemMessage implements Message {
   @Nonnull String role = "system";
 
   /** The content of the message. */
-  MessageContent content;
-
-  @Nonnull
-  @Override
-  public String content() {
-    return MessageContent.toString(content);
-  }
+  @Nonnull MessageContent content;
 
   @Override
   @Nonnull
-  public MessageContent getContent() {
-    return content != null ? content : new MessageContentSingle("");
+  public MessageContent content() {
+    return content;
   }
 
   public SystemMessage(String singleMessage) {
     content = new MessageContentSingle(singleMessage);
   }
 
-  public SystemMessage(MessageContent messageContent) {
+  SystemMessage(MessageContent messageContent) {
+    if (!(messageContent instanceof MessageContentSingle)) {
+      ((MessageContentMulti) messageContent)
+          .multiContentList().stream()
+              .filter(mCMC -> !(mCMC instanceof MultiMessageTextContent))
+              .findAny()
+              .ifPresent(
+                  mCMC -> {
+                    throw new IllegalArgumentException(
+                        "Only TextContent is supported for SystemMessage");
+                  });
+    }
```

Comment:
(Minor)
I would leave it out. Prematurely throwing exceptions at runtime is an unnecessary risk. There is no benefit for developers at design time right? The user would still do the regular casts.
Instead I would just add a javadoc or comment, that only instances of `MultiMessageTextContent` are expected.